crypto/s2n_hmac.c

```diff
@@ -294,32 +311,41 @@ int s2n_hmac_digest_two_compression_rounds(struct s2n_hmac_state *state, void *o
 
 int s2n_hmac_free(struct s2n_hmac_state *state)
 {
+    if (!state) { return S2N_SUCCESS; }
     GUARD(s2n_hash_free(&state->inner));
     GUARD(s2n_hash_free(&state->inner_just_key));
     GUARD(s2n_hash_free(&state->outer));
     GUARD(s2n_hash_free(&state->outer_just_key));
 
-    return 0;
+    return S2N_SUCCESS;
 }
 
 int s2n_hmac_reset(struct s2n_hmac_state *state)
 {
+    PRECONDITION_POSIX(s2n_hmac_state_validate(state));
     GUARD(s2n_hash_copy(&state->inner, &state->inner_just_key));
 
     uint64_t bytes_in_hash;
     GUARD(s2n_hash_get_currently_in_hash_total(&state->inner, &bytes_in_hash));
+    ENSURE_POSIX(state->hash_block_size != 0, S2N_ERR_INTEGER_OVERFLOW);
```

Comment:
This looks a bit odd to me...

Should this be an overflow error or just a precondition violation?

Reply:
You're right, I have updated it as suggested.